src/Domain/CoordinateMaps/CubicScale.cpp

```diff
@@ -101,168 +168,258 @@ boost::optional<std::array<tt::remove_cvref_wrap_t<T>, 1>> CubicScale::inverse(
-  auto inv_jac{
-      make_with_value<tnsr::Ij<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame>>(
-          dereference_wrapper(source_coords[0]), 0.0)};
 
-  get<0, 0>(inv_jac) = 1.0 / (a_of_t +
-                              3.0 * (b_of_t - a_of_t) *
-                                  square(source_coords[0] / outer_boundary_));
+  if (functions_of_time_equal_) {
+    // optimization for linear radial scaling
+    auto inv_jac{make_with_value<
+        tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame>>(
+        dereference_wrapper(source_coords[0]), 0.0)};
+    const double one_over_a = 1.0 / a_of_t;
+    for (size_t i = 0; i < Dim; ++i) {
+      inv_jac.get(i, i) = one_over_a;
+    }
+    return inv_jac;
+  }
 
-  return inv_jac;
-}
+  const auto b_of_t = map_list.at(f_of_t_b_)->func(time)[0][0];
 
-template <typename T>
-tnsr::Iaa<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame> CubicScale::hessian(
-    const std::array<T, 1>& source_coords, const double time,
-    const std::unordered_map<
-        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
-        map_list) const noexcept {
-  const auto a_of_t_and_derivs =
-      map_list.at(f_of_t_a_)->func_and_2_derivs(time);
-  const auto b_of_t_and_derivs =
-      map_list.at(f_of_t_b_)->func_and_2_derivs(time);
-
-  auto result{
-      make_with_value<tnsr::Iaa<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame>>(
-          dereference_wrapper(source_coords[0]), 0.0)};
-  // time-time
-  get<0, 0, 0>(result) = a_of_t_and_derivs[2][0] * source_coords[0] +
-                         (b_of_t_and_derivs[2][0] - a_of_t_and_derivs[2][0]) *
-                             cube(source_coords[0]) / square(outer_boundary_);
-  // time-space
-  get<0, 0, 1>(result) =
-      a_of_t_and_derivs[1][0] +
-      3.0 * (b_of_t_and_derivs[1][0] - a_of_t_and_derivs[1][0]) *
-          square(source_coords[0] / outer_boundary_);
-  // space-space
-  get<0, 1, 1>(result) = 6.0 *
-                         (b_of_t_and_derivs[0][0] - a_of_t_and_derivs[0][0]) *
-                         source_coords[0] / square(outer_boundary_);
+  tt::remove_cvref_wrap_t<T> r_squared =
+      square(dereference_wrapper(source_coords[0]));
+  for (size_t i = 1; i < Dim; ++i) {
+    r_squared += square(dereference_wrapper(gsl::at(source_coords, i)));
+  }
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame> inv_jac{};
+  get<0, 0>(inv_jac) =
+      1.0 / (a_of_t + (b_of_t - a_of_t) / square(outer_boundary_) * r_squared);
+  for (size_t i = 1; i < Dim; ++i) {
+    inv_jac.get(i, i) = get<0, 0>(inv_jac);
+  }
+  if (Dim == 1) {
+    get<0, 0>(inv_jac) *= (1.0 + (-2.0 * (b_of_t - a_of_t)) /
+                                     (a_of_t * square(outer_boundary_) +
+                                      3.0 * (b_of_t - a_of_t) * r_squared) *
+                                     square(source_coords[0]));
+  } else {
+    // Reuse r^2 allocation
+    r_squared = (-2.0 * (b_of_t - a_of_t)) /
```

Comment:
I confirmed that this is the correct inverse, but I think it will be more maintainable with comments